tests/service.rs

```diff
@@ -24,8 +24,38 @@ fn service_should_be_started_successfully() {
                 - key: kubernetes.io/arch
                   operator: Equal
                   value: stackable-linux
-        "},
+        "}),
     );
 
     client.verify_pod_condition(&pod, "Ready");
 }
+
+#[test]
+fn restart_after_ungraceful_shutdown_should_succeed() {
+    let client = TestKubeClient::new();
+
+    setup_repository(&client);
+
+    let pod_spec = with_unique_name(indoc! {"
+        apiVersion: v1
+        kind: Pod
+        metadata:
+          name: agent-service-integration-test-restart
+        spec:
+          containers:
+            - name: nostop-service
+              image: nostop-service:1.0.1
+              command:
+                - nostop-service-1.0.1/start.sh
+          tolerations:
+            - key: kubernetes.io/arch
+              operator: Equal
+              value: stackable-linux
+          terminationGracePeriodSeconds: 2
+    "});
+
+    for _ in 1..=2 {
```

Comment:
Just to ensure my understanding is correct: this relies on the temporaryresource being deleted when it goes out of scope after every iteration of the loop?

Reply:
Exactly.